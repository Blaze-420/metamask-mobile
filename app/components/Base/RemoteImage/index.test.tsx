import React from 'react';
import RemoteImage from './';
import { getFormattedIpfsUrl } from '@metamask/assets-controllers';
import { act, render, waitFor } from '@testing-library/react-native';
import { useSelector } from 'react-redux';
import { backgroundState } from '../../../util/test/initial-root-state';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn().mockImplementation(() => 'https://dweb.link/ipfs/'),
}));

jest.mock('../../../components/hooks/useIpfsGateway', () => jest.fn());

jest.mock('@metamask/assets-controllers', () => ({
  getFormattedIpfsUrl: jest.fn(),
}));

jest.mock('../../../util/networks', () => ({
  ...jest.requireActual('../../../util/networks'),
  isSolanaMainnet: jest.fn(),
}));

const mockGetFormattedIpfsUrl = getFormattedIpfsUrl as jest.Mock;

describe('RemoteImage', () => {
  it('renders the remote svg image snapshot', () => {
    const wrapper = render(
      <RemoteImage
        source={{
          uri: 'https://raw.githubusercontent.com/MetaMask/contract-metadata/master/images/dai.svg',
        }}
      />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders the remote png image snapshot', () => {
    const wrapper = render(
      <RemoteImage
        source={{
          uri: 'https://s3.amazonaws.com/airswap-token-images/OXT.png',
        }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the remote ipfs image snapshot', async () => {
    const testIpfsUri = 'ipfs://QmeE94srcYV9WwJb1p42eM4zncdLUai2N9zmMxxukoEQ23';
    mockGetFormattedIpfsUrl.mockResolvedValue(testIpfsUri);
    const { toJSON, getByTestId } = render(
      <RemoteImage
        source={{
          uri: testIpfsUri,
        }}
      />,
    );
    await waitFor(() =>
      expect(getByTestId('mock-expo-image')).toHaveProp('source', {
        uri: testIpfsUri,
      }),
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders remote solana image and badge snapshot', async () => {
    // @ts-expect-error - useSelector is mocked in the top of the file
    useSelector.mockImplementation((selector) => {
      const mockState = {
        engine: {
          backgroundState: {
            ...backgroundState,
            MultichainNetworkController: {
              ...backgroundState.MultichainNetworkController,
              isEvmSelected: false,
            },
          },
        },
      };
      return selector(mockState);
    });

    const wrapper = render(
      <RemoteImage
        fadeIn
        isTokenImage
        source={{
          uri: 'https://example.com/token.png',
        }}
      />,
    );

    // eslint-disable-next-line no-empty-function
    await act(async () => {});
    expect(wrapper).toMatchSnapshot();
  });
});
