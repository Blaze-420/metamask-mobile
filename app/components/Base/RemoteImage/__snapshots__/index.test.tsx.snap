// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoteImage renders remote solana image and badge snapshot 1`] = `
<View
  useNativeDriver={true}
>
  <View
    onLoadEnd={[Function]}
  >
    <View
      onLayout={[Function]}
      style={
        {
          "alignSelf": "flex-start",
          "position": "relative",
        }
      }
      testID="badge-wrapper-badge"
    >
      <View>
        <View>
          <Image
            fadeIn={true}
            isTokenImage={true}
            onError={[Function]}
            resizeMode="cover"
            source={
              {
                "uri": "ipfs://QmeE94srcYV9WwJb1p42eM4zncdLUai2N9zmMxxukoEQ23",
              }
            }
            style={
              {
                "borderRadius": 8,
                "height": "100%",
                "width": "100%",
              }
            }
            testID="mock-expo-image"
          />
        </View>
      </View>
      <View
        style={
          {
            "position": "absolute",
            "right": -4,
            "top": -4,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={{}}
          testID="badgenetwork"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#f3f5f9",
                "borderColor": "#ffffff",
                "borderRadius": 8,
                "borderWidth": 1,
                "height": 16,
                "justifyContent": "center",
                "overflow": "hidden",
                "shadowColor": "#0000001a",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 4,
                "width": 16,
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#121314",
                  "fontFamily": "CentraNo1-Book",
                  "fontSize": 10,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": undefined,
                }
              }
            >
              S
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        [
          {},
          {
            "backgroundColor": "#eee",
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`RemoteImage renders the remote ipfs image snapshot 1`] = `
<Image
  onError={[Function]}
  source={
    {
      "uri": "ipfs://QmeE94srcYV9WwJb1p42eM4zncdLUai2N9zmMxxukoEQ23",
    }
  }
  testID="mock-expo-image"
/>
`;

exports[`RemoteImage renders the remote png image snapshot 1`] = `
<Image
  onError={[Function]}
  source={
    {
      "uri": "https://s3.amazonaws.com/airswap-token-images/OXT.png",
    }
  }
  testID="mock-expo-image"
/>
`;

exports[`RemoteImage renders the remote svg image snapshot 1`] = `
<View
  style={
    {
      "overflow": "hidden",
    }
  }
>
  <Image
    contentFit="contain"
    source={
      {
        "uri": "https://raw.githubusercontent.com/MetaMask/contract-metadata/master/images/dai.svg",
      }
    }
    style={
      {
        "aspectRatio": 1,
        "height": "100%",
        "width": "100%",
      }
    }
    testID="mock-expo-image"
  />
</View>
`;
